Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -3561,6 +3571,29 @@ void RenderBox::computePreferredLogicalWidths()
 
 void RenderBox::computePreferredLogicalWidths(const Length& minLogicalWidth, const Length& maxLogicalWidth, LayoutUnit borderAndPaddingLogicalWidth)
 {
+
+    auto usedMaxLogicalWidth = [&] {
+        // FIXME: We should be able to handle other values for the max logical width here.
+        if (maxLogicalWidth.isFixed())
+            return adjustContentBoxLogicalWidthForBoxSizing(maxLogicalWidth);
+
+        if (maxLogicalWidth.isMinContent()) {
+            if (!shouldComputePreferredLogicalWidthsFromStyle())
+                return m_minPreferredLogicalWidth;
+
+            static constexpr LayoutUnit s_noBorderAndPaddingLogicalWidth;
+            return computeIntrinsicLogicalWidthUsing(maxLogicalWidth, availableLogicalWidth(), s_noBorderAndPaddingLogicalWidth);
```

Comment:
I think `computeIntrinsicLogicalWidthUsing(maxLogicalWidth, availableLogicalWidth(), { });` is a more readable way to express passing in default values even if it has some runtime overhead over the static variable which I don't think really matters here. -or assign a default value at the function declaration.